### PR TITLE
Fixed casing issue causing CloudError in Azure clients to always be null #668

### PR DESCRIPTION
**Pull request by stankovski**
#668

### CODE REVIEW

File: ClientRuntimes/CSharp/Microsoft.Rest.ClientRuntime.Azure/CloudErrorJsonConverter.cs [hunk 1]
```diff
@@ -49,9 +51,10 @@ public override bool CanConvert(Type objectType)
             }
 
             JObject jObject = JObject.Load(reader);
-            if (jObject.Property(ErrorNode) != null)
+            JProperty errorObject = jObject.Properties().FirstOrDefault(p => ErrorNode.Equals(p.Name, StringComparison.OrdinalIgnoreCase));
```
Comment (markcowl) [inline ClientRuntimes/CSharp/Microsoft.Rest.ClientRuntime.Azure/CloudErrorJsonConverter.cs]: break line

File: ClientRuntimes/CSharp/Microsoft.Rest.ClientRuntime.Azure.Tests/CloudErrorJsonConverterTest.cs [hunk 2]
```diff
@@ -47,6 +48,34 @@ public void TestCloudErrorDeserialization()
         }
 
         [Fact]
+        public void TestCloudErrorWithDifferentCasing()
+        {
+            var expected = "{\"Error\":{\"Code\":\"CatalogObjectNotFound\",\"Message\":\"datainsights.blah doesn't exist.\"}} ";
```
Comment (begoldsm) [inline ClientRuntimes/CSharp/Microsoft.Rest.ClientRuntime.Azure.Tests/CloudErrorJsonConverterTest.cs]: Shouldn't you also have one with lower case to ensure the casing is fixed?

**Comment by markcowl** [comment]
@stankovski one nit, otherwise lgtm when the build passes

File: ClientRuntimes/CSharp/Microsoft.Rest.ClientRuntime.Azure.Tests/CloudErrorJsonConverterTest.cs (hunk 2 above)
Comment (stankovski) [inline ClientRuntimes/CSharp/Microsoft.Rest.ClientRuntime.Azure.Tests/CloudErrorJsonConverterTest.cs]: Yep, test TestCloudErrorDeserialization() few lines above has lower case.

**Pull request merged**